### PR TITLE
Clean up obsolete code and improve structure

### PR DESCRIPTION
**Pull request by bmuschko**
- Travis CI should not run default `assemble` task. It's pointless.
- Remove unnecessary code in generated `build.gradle` file.
- Automatic setup of samples code/output directory structure.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-import org.apache.tools.ant.filters.*
-
-guide {
-    repoPath 'gradle-guides/@@GUIDE_SLUG@@'
-    mainAuthor 'EDIT build.gradle TO ADD AUTHOR'
-
-    // Uncomment this line to add support authors
-    // supAuthors 'Grumpy Gradlephant', 'Jane Doe'
+configure<org.gradle.guides.GuidesExtension> {
+    repoPath = 'gradle-guides/@@GUIDE_SLUG@@'
```
Comment (eriwen) [inline build.gradle]: Do single quotes work here?

Comment (eriwen) [reply build.gradle]: We should stick to double quotes for consistency even if it does.

Comment (bmuschko) [reply build.gradle]: Either a copy-paste issue or doesn't work. Correct.

File: build.gradle [hunk 2]
```diff
@@ -1,21 +1,20 @@
 plugins {
-    id "org.gradle.guides.getting-started" version "0.11.5"
+    id("org.gradle.guides.getting-started") version "0.11.5"
```
Comment (eriwen) [inline build.gradle]: D'oh! This should be build.gradle.kts

**Pull request merged**